dataloader.lua

```diff
@@ -451,30 +448,11 @@ function dataloader.getIndexOption(self, inds, params, dtype)
         optionIn = optionIn:view(optInds:size(1), optInds:size(2),
                                                 optInds:size(3), -1);
         optionIn = optionIn[{{}, {}, {}, {1, maxOptLen}}];
-        if dtype == 'test' then
-            -- convert optionIn to keep options for n-th (last) round and zeros for other rounds
-            local optionInTest = torch.zeros(optionIn:size(1), 10, optionIn:size(3), optionIn:size(4)):int()
```

Comment:
@abhshkdz `indVector` had a lot of zeros from preprocessing script for empty options - and lua is one indexed. So I avoided zeros by storing only one round options `visdial_data.h5` (which meant the dimension of that entry to be different for `test` and (`train`, `val`). I made it one-indexed anyway, it now fetches the first answer for test split intermediate dialog rounds. We don't care how they are ranked in inference mode.